src/lazy_execution/values.rs

```diff
@@ -0,0 +1,354 @@
+// -*- coding: utf-8 -*-
+// ------------------------------------------------------------------------------------------------
+// Copyright © 2021, tree-sitter authors.
+// Licensed under either of Apache License, Version 2.0, or MIT license, at your option.
+// Please see the LICENSE-APACHE or LICENSE-MIT files in this distribution for license details.
+// ------------------------------------------------------------------------------------------------
+
+//! Defines values for lazy DSL evaluation
+
+use log::trace;
+
+use std::convert::From;
+use std::fmt;
+
+use crate::execution::ExecutionError;
+use crate::graph;
+use crate::graph::DisplayWithGraph;
+use crate::graph::Graph;
+use crate::graph::GraphNodeRef;
+use crate::graph::SyntaxNodeRef;
+use crate::Context;
+use crate::DisplayWithContext;
+use crate::Identifier;
+
+use super::store::*;
+use super::DisplayWithContextAndGraph;
+use super::EvaluationContext;
+
+/// Lazy values
+#[derive(Clone, Debug)]
+pub enum Value {
```

Comment:
This might seem like a cumbersome suggestion, but can we rename this to `LazyValue` (ditto the other types in this module)?  I actually didn't realize until this point in the PR that all of the references to `Value` earlier on were referring to a new type.

Unless the intent is that these types will _replace_ the equivalently named ones in the `graph` module once we remove the old evaluation strategy?  If that's the case, then we can deal with a bit of confusion in the meantime.  But if both will exist even once lazy evaluation is the only strategy that's implemented, then I think it's important for the names to clarify the difference between a lazy value and a forced strict value.

Reply:
> Unless the intent is that these types will replace the equivalently named ones in the graph module once we remove the old evaluation strategy?

No, they'll always be separate. I can prefix them to make the code more clear.